### DISCUSSION TITLE
Need help troubleshooting SERVFAIL

### DISCUSSION THREAD

**Discussion by neilio** [AdguardTeam/AdGuardHome#4867] [Q&A]
I'm trying to run AGH on my Mac mini. When I reboot my system AGH doesn't work unless I stop and restart it up to three times. Before then, every request returns SERVFAIL.

System / version
macOS 12.5.1
Mac mini M1
AdGuard Home v0.108.0-b.13

Setup
Downloaded, unzipped, and then ran -s install from the command line
On one machine, DHCP managed by router

Does anyone have any advice on where I can look to figure out what's going wrong?

**Comment by ainar-g** [comment]
[Verbose logs](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog) of AdGuard Home would be a good place to start.  `dmesg` and other system log files as well, to diagnose network issues and, perhaps, OS bugs.

**Reply by neilio** [reply]
Here's a verbose log I just captured. I launched AGH, tried a domain I tried earlier (which I assume was cached as it worked), and then ran nslookup for foxiness.com. Then I shut AGH down. I don't see any record of foxiness.com in the verbose log at all, which is odd.
[adglog.txt](https://github.com/AdguardTeam/AdGuardHome/files/9483019/adglog.txt)

**Reply by ainar-g** [reply]
That most likely means that your system isn't setup to use AdGuard Home as its only DNS server.  You should recheck your configuration.

**Reply by neilio** [reply]
> That most likely means that your system isn't setup to use AdGuard Home as its only DNS server. You should recheck your configuration.

Does this mean setting 127.0.0.1 as the DNS server for the machine AGH is running on? I have that already configured in system settings for both network adapters (wifi and ethernet).

**Reply by ainar-g** [reply]
There are several other places, including the DNS settings that your router sends as well as the fact that modern browsers will often use Google or Cloudflare DoH by default, and if you want to use your own DNS, you'll have to disable that.  My first guess is that the settings that your router is sending overwrite the ones you set, and that there is an option to ignore the router's DNS settings somewhere in the UI.

**Comment by neilio** [comment]
I figured out the problem – at the bottom of my Upstream DNS Servers I had three different servers with two of them commented out. Once I removed the commented lines, everything works as expected.

```
# tls://account.dns.controld.com
https://dns.nextdns.io/account
# quic://account.d.adguard-dns.com
```
